api/client/swarm/opts.go

```diff
@@ -25,25 +34,36 @@ var (
 // NodeAddrOption is a pflag.Value for listen and remote addresses
 type NodeAddrOption struct {
 	addr string
+	port uint16
 }
 
 // String prints the representation of this flag
 func (a *NodeAddrOption) String() string {
-	return a.addr
+	return a.Value()
 }
 
 // Set the value for this flag
 func (a *NodeAddrOption) Set(value string) error {
 	if !strings.Contains(value, ":") {
-		return fmt.Errorf("Invalid url, a host and port are required")
+		a.addr = value
+		return nil
 	}
 
 	parts := strings.Split(value, ":")
```

Comment:
Thanks, I thought this split felt wrong, but my quick search for such a function didn't turn up anything. Now I know where to find it.